### PR TITLE
Testsforproject

### PR DESCRIPTION
**Pull request by Dima23-ops5**

### CODE REVIEW

File: src/test/java/mate/academy/springbootintro/controllers/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/book/remove-all-books.sql"));
+        }
+    }
+
+    @WithMockUser(username = "user", roles = {"USER"})
+    @Test
+    @DisplayName("Get all books")
+    public void getAll_FindAllBooks_Correct() throws Exception {
+        BookDto book1 = new BookDto();
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/springbootintro/controllers/BookControllerTest.java]: don't use numbers for naming

File: src/test/java/mate/academy/springbootintro/controllers/BookControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+
+        assertNotNull(actual);
+        assertEquals(excepted.getId(), actual.getId());
+        assertEquals(excepted.getTitle(), actual.getTitle());
+        assertEquals(excepted.getAuthor(), actual.getAuthor());
+        assertEquals(excepted.getIsbn(), actual.getIsbn());
+        assertEquals(excepted.getPrice(), actual.getPrice());
+        assertEquals(excepted.getDescription(), actual.getDescription());
+        assertEquals(excepted.getCoverImage(), actual.getCoverImage());
+    }
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/springbootintro/controllers/BookControllerTest.java:286]: add not valid cases

File: src/test/java/mate/academy/springbootintro/controllers/CategoryControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+                get("/categories/4/books").contentType(MediaType.APPLICATION_JSON)
+        ).andExpect(status().isOk())
+                .andReturn();
+        List<BookDtoWithoutCategoryIds> actual = Arrays.stream(objectMapper
+                .readValue(result.getResponse().getContentAsString(),
+                        BookDtoWithoutCategoryIds[].class)).toList();
+
+        assertEquals(1, actual.size());
+        assertEquals(actual, excepted);
+    }
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/springbootintro/controllers/CategoryControllerTest.java:231]: same here

File: src/test/java/mate/academy/springbootintro/service/BookServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    private BookMapper bookMapper;
+    @Mock
+    private BookSpecificationBuilder bookSpecificationBuilder;
+    @InjectMocks
+    private BookServiceImpl bookService;
+
+    @Test
+    @DisplayName("Find all books")
+    public void findAll_ThreeBooks_Correct() {
+
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/springbootintro/service/BookServiceImplTest.java]:
```suggestion
```

File: src/test/java/mate/academy/springbootintro/service/BookServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+        Book book = createTestBook();
+
+        BookDto bookDto = createBookDto();
+
+        Page<Book> page = new PageImpl<>(List.of(book));
+        Pageable pageable = PageRequest.of(0, 10);
+
+        Mockito.when(bookMapper.toDto(book)).thenReturn(bookDto);
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/springbootintro/service/BookServiceImplTest.java]: use static imports

File: src/test/java/mate/academy/springbootintro/service/BookServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+
+        String expected = "Cannot find book with id: " + bookId;
+        String actual = exception.getMessage();
+        assertEquals(actual, expected);
+    }
+
+    @Test
+    @DisplayName("Update book")
+    public void updateBook_UpdateBookWithExistedId_Correct() {
+
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/springbootintro/service/BookServiceImplTest.java]:
```suggestion
```

File: src/test/java/mate/academy/springbootintro/service/BookServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+        Mockito.when(bookRepository.save(book)).thenReturn(book);
+
+        BookDto actual = bookService.save(requestDto);
+        assertNotNull(actual);
+        assertEquals(actual, bookDto);
+    }
+
+    @Test
+    @DisplayName("Get book with id that doesn't exist")
+    public void getBook_GetBookById_False() {
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/springbootintro/service/BookServiceImplTest.java]:
```suggestion
    public void getBook_GetBookByNotValidId_NotOk) {
```

Comment (Elena-Bruyako) [reply src/test/java/mate/academy/springbootintro/service/BookServiceImplTest.java]: or findById_WithInvalidId_ShouldThrowException

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

check all test according to the comments (pay attention to naming)

File: coverage-error.log [hunk 8]
```diff
@@ -0,0 +1,21 @@
+[2024.08.23 10:13:22] (Coverage ERROR): Failed to parse agent arguments
```
Comment (okuzan) [inline coverage-error.log]: why is this file here?

**Review by okuzan — CHANGES_REQUESTED** [review]
image
application misspelled

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request closed without merging**